mongo/options/clientencryptionoptions.go

```diff
@@ -43,7 +82,47 @@ func MergeClientEncryptionOptions(opts ...*ClientEncryptionOptions) *ClientEncry
 		if opt.KmsProviders != nil {
 			ceo.KmsProviders = opt.KmsProviders
 		}
+		if opt.TLSConfig != nil {
+			ceo.TLSConfig = opt.TLSConfig
+		}
 	}
 
 	return ceo
 }
+
+// Used by both ClientEncryptionOptions.SetTLSOptions() and AutoEncryptionOptions.SetTLSOptions()
+func applyTLSOptions(tlsOpts map[string]map[string]interface{}, tlsConfigs map[string]*tls.Config) (map[string]*tls.Config, error) {
+	for provider, opts := range tlsOpts {
+		cfg := &tls.Config{MinVersion: tls.VersionTLS12}
+		
+		for cert := range opts {
+			var err error
+			switch cert {
+			case "tlsCertificateKeyFile", "sslClientCertificateKeyFile":
+				clientCertPath, _ := opts[cert].(string)
```

Comment:
This will silently continue if `opts[cert]` is not of type `string`. You could save the "ok" value and error if it's false.
```suggestion
				clientCertPath, ok := opts[cert].(string)
				if !ok {
					return tlsConfigs, fmt.Errorf("expected %q value to be of type string, got %T", cert, opts[cert])
				}
```

Reply:
Good catch thank you!